### ISSUE TITLE
unable to see color of colorized objects behind x-rayed objects

### ISSUE THREAD

**Issue by ohadgilboa888** [xeokit/xeokit-sdk#864]
**Describe the bug**
unable to see color of colorized objects behind x-rayed objects

**To Reproduce**
1. set object with color (setObjectsColorized)
2. set object on xray mode (setObjectsXRayed)
3. view the xrayed object in front of the colorized object

**Expected behavior**
object's color should be seen through the xrayed object

**Screenshots**
green objects
<img width="649" alt="image" src="https://user-images.githubusercontent.com/91824391/170483200-1ab2ae2c-a701-4d75-9006-e1a9d016c04d.png">
green objects behind layer of xrayed objects 
<img width="691" alt="image" src="https://user-images.githubusercontent.com/91824391/170483305-cc6490ee-cf17-4b5c-aaf2-13ca3fc1c63f.png">

**Desktop (please complete the following information):**
 - Mac
 - Browser: chrome
 
didn't work with objects with low opacity (instead of xrayed)

**Comment by xeolabs** [comment]
I'm not able to reproduce this issue so far - see details below. Could you provide a way to reproduce, perhaps a model?

Live example (gh-pages sometimes takes ages to update, so live example may not appear soon):

https://xeokit.github.io/xeokit-sdk/emphasis_colorize_throughXRay.html
https://github.com/xeokit/xeokit-sdk/blob/master/examples/emphasis_colorize_throughXRay.html

Result:

![Screenshot from 2022-06-17 13-25-06](https://user-images.githubusercontent.com/83100/174289436-73b8dc85-f007-4499-a00d-0818c70571f5.png)

Code:

````javascript
    import {Viewer, VBOSceneModel} from "./../dist/xeokit-sdk.es.js";

    const viewer = new Viewer({
        canvasId: "myCanvas",
        transparent: true,
        pbrEnabled: true
    });

    viewer.scene.camera.eye = [0, 0, -15];
    viewer.scene.camera.look = [0, 0, 0];
    viewer.scene.camera.up = [0, 1, 0];


    viewer.scene.highlightMaterial.alpha = 0.4;

    const vboSceneModel = new VBOSceneModel(viewer.scene, {
        id: "testModel",
        edges: true
    });

    vboSceneModel.createGeometry({
        id: "myBoxGeometry",
        primitive: "triangles",
        positions: [
            1, 1, 1, -1, 1, 1, -1, -1, 1, 1, -1, 1, // v0-v1-v2-v3 front
            1, 1, 1, 1, -1, 1, 1, -1, -1, 1, 1, -1, // v0-v3-v4-v1 right
            1, 1, 1, 1, 1, -1, -1, 1, -1, -1, 1, 1, // v0-v1-v6-v1 top
            -1, 1, 1, -1, 1, -1, -1, -1, -1, -1, -1, 1, // v1-v6-v7-v2 left
            -1, -1, -1, 1, -1, -1, 1, -1, 1, -1, -1, 1,// v7-v4-v3-v2 bottom
            1, -1, -1, -1, -1, -1, -1, 1, -1, 1, 1, -1 // v4-v7-v6-v1 back
        ],
        normals: [
            0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1,// v0-v1-v2-v3 front
            1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0, 0,// v0-v3-v4-v5 right
            0, 1, 0, 0, 1, 0, 0, 1, 0, 0, 1, 0,// v0-v5-v6-v1 top
            -1, 0, 0, -1, 0, 0, -1, 0, 0, -1, 0, 0,// v1-v6-v7-v2 left
            0, -1, 0, 0, -1, 0, 0, -1, 0, 0, -1, 0,// v7-v4-v3-v2 bottom
            0, 0, -1, 0, 0, -1, 0, 0, -1, 0, 0, -1// v4-v7-v6-v5 back
        ],
        indices: [
            0, 1, 2, 0, 2, 3,            // front
            4, 5, 6, 4, 6, 7,            // right
            8, 9, 10, 8, 10, 11,         // top
            12, 13, 14, 12, 14, 15,      // left
            16, 17, 18, 16, 18, 19,      // bottom
            20, 21, 22, 20, 22, 23
        ]
    });

    vboSceneModel.createMesh({
        id: "innerBoxMesh",
        geometryId: "myBoxGeometry",
        position: [0, 0, 0],
        scale: [1, 1, 1],
        color: [1, 0.3, 0.3]
    });

    const innerBox = vboSceneModel.createEntity({
        id: "innerBox",
        meshIds: ["innerBoxMesh"]
    });

    vboSceneModel.createMesh({
        id: "outerBoxMesh",
        geometryId: "myBoxGeometry",
        position: [0, 0, 0],
        scale: [1.5, 1.5, 1.5],
        color: [0.3, 1.0, 0.3],
        opacity: 0.5
    });

    const outerBox = vboSceneModel.createEntity({
        id: "outerBox",
        meshIds: ["outerBoxMesh"]
    });

    vboSceneModel.finalize();

    innerBox.colorize = [1, 0, 0];
    innerBox.opacity = 0.3;

    outerBox.xrayed = true;
````

**Comment by xeolabs** [comment]
Fixed in 2.3.0-beta-16

Test case: 
https://xeokit.github.io/xeokit-sdk/examples/#emphasis_colorize_throughXRay

Before fix:

![Screenshot from 2022-06-17 13-25-06](https://user-images.githubusercontent.com/83100/174291271-85ae4e3a-4f79-4fc0-960f-6894d055ad2f.png)

After fix:
![Screenshot from 2022-06-17 13-40-00](https://user-images.githubusercontent.com/83100/174291481-14561d19-c3c5-42fd-90b3-14088cb50944.png)

**Comment by xeolabs** [comment]
Fixed in 2.3.0-beta-16

Test case: 
https://xeokit.github.io/xeokit-sdk/examples/#emphasis_colorize_throughXRay
https://github.com/xeokit/xeokit-sdk/blob/master/examples/emphasis_colorize_throughXRay.html

Before fix:

![Screenshot from 2022-06-17 13-25-06](https://user-images.githubusercontent.com/83100/174291271-85ae4e3a-4f79-4fc0-960f-6894d055ad2f.png)

After fix:
![Screenshot from 2022-06-17 13-40-00](https://user-images.githubusercontent.com/83100/174291481-14561d19-c3c5-42fd-90b3-14088cb50944.png)
